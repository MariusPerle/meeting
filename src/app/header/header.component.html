<div class="row header">
  <div class="col info">
    <span (click)="infoDialog.showModal()" class="data">
      Learn about the data policy
      <i class="bi bi-info-circle"></i>
    </span>
    <dialog #infoDialog>
      <p>All data is stored locally in your browser</p>
      <div class="interactions">
        <span>
          <a href="https://www.w3schools.com/jsref/prop_win_localstorage.asp" target="_blank"
                 rel="noopener noreferrer">learn more</a>
        </span>
        <button (click)="infoDialog.close()" aria-label="Close">Close</button>
      </div>
    </dialog>
  </div>
  <div class="col"><h1>{{ title$ | async }}</h1></div>
  <div class="col options">
    <i class="bi bi-list" (click)="interactions.showPopover()"></i>
    <div class="navigation-popover" #interactions popover>
      <div (click)="navigate('/')">Home</div>
      <div (click)="navigate('/meeting/list')">Show History</div>
      <div (click)="navigate('/team')">Team</div>
      <div>
        <daily-reset></daily-reset>
      </div>
    </div>
  </div>
</div>