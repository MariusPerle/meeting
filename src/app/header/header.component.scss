:host .header {
    margin: 0;

    .info {
        display: flex;
        align-items: center;

        .data {
            cursor: pointer;

            .bi {
                margin-left: 1ch;
                margin-right: 3ch;
            }
        }

        dialog {
            .interactions {
                justify-content: space-between;
            }
        }

    }

    h1 {
        text-align: center;
    }

    .options {
        display: flex;
        align-items: center;
        justify-content: flex-end;
        position: relative;

        .bi-list {
            cursor: pointer;
        }

        .navigation-popover {
            cursor: pointer;
            margin: 0;
            padding: 0;
            position: absolute;
            left: unset;
            right: 35px;
            top: 15px;
            border: none;
            border-radius: 5px;
            box-shadow: 0 0 5px #727272;
            &>div {
                padding: 0.4ch;

                &+div {
                    border-top: 1px solid #727272;
                }
            }
        }
    }
}