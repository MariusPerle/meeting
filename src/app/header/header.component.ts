import { Component, ElementRef, inject, viewChild } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ResetComponent } from './reset/reset.component';
import { Router } from '@angular/router';
import { AppService } from '../app.service';

@Component({
    selector: 'meeting-header',
    imports: [CommonModule, ResetComponent],
    templateUrl: './header.component.html',
    styleUrl: './header.component.scss',
})
export class HeaderComponent {
    private readonly appService = inject(AppService);
    private readonly router = inject(Router);

    title$ = this.appService.title$;
    interactions = viewChild<ElementRef<HTMLDivElement>>('interactions');

    navigate(path: string) {
        this.router
            .navigate([path])
            .then(() => this.interactions()?.nativeElement.hidePopover());
    }
}
