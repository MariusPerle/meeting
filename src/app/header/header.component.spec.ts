import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HeaderComponent } from './header.component';
import { Router } from '@angular/router';

describe('HeaderComponent', () => {
    let component: HeaderComponent;
    let fixture: ComponentFixture<HeaderComponent>;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [HeaderComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(HeaderComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('sould render corrently', () => {
        expect(component).toBeTruthy();
        expect(fixture).toMatchSnapshot();
    });

    it('should open and close popover with navigation', () => {
        const popover: HTMLDivElement = fixture.nativeElement.querySelector(
            '.navigation-popover',
        );
        popover.showPopover = jest.fn();
        popover.hidePopover = jest.fn();
        fixture.nativeElement.querySelector('.bi-list').click();
        expect(popover.showPopover).toHaveBeenCalled();

        jest.spyOn(
            (component as unknown as { router: Router }).router,
            'navigate',
        );

        fixture.nativeElement.querySelector('.navigation-popover div').click();
        expect(popover.hidePopover).toHaveBeenCalled();

        fixture.nativeElement.querySelector('.navigation-popover div').click();
        expect(
            (component as unknown as { router: Router }).router.navigate,
        ).toHaveBeenCalledWith(['/']);
    });
});
