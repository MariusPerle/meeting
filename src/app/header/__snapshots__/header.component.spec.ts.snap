// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent sould render corrently 1`] = `
<daily-header
  appService={[Function AppService]}
  interactions={[Function Function]}
  router={[Function _Router]}
  title$={[Function BehaviorSubject]}
>
  <div
    class="row header"
  >
    <div
      class="col info"
    >
      <span
        class="data"
      >
         Learn about the data policy 
        <i
          class="bi bi-info-circle"
        />
      </span>
      <dialog>
        <p>
          All data is stored locally in your browser
        </p>
        <div
          class="interactions"
        >
          <span>
            <a
              href="https://www.w3schools.com/jsref/prop_win_localstorage.asp"
              rel="noopener noreferrer"
              target="_blank"
            >
              learn more
            </a>
          </span>
          <button
            aria-label="Close"
          >
            Close
          </button>
        </div>
      </dialog>
    </div>
    <div
      class="col"
    >
      <h1>
      </h1>
    </div>
    <div
      class="col options"
    >
      <i
        class="bi bi-list"
      />
      <div
        class="navigation-popover"
        popover=""
      >
        <div>
          Home
        </div>
        <div>
          Show History
        </div>
        <div>
          Team
        </div>
        <div>
          <daily-reset>
            <div>
              Delete all Data
            </div>
            <dialog>
              <div>
                 Are you sure you want to delete all data? 
              </div>
              <div
                class="interactions"
              >
                <button
                  aria-label="Remove"
                  class="confirm"
                >
                  Confirm
                </button>
                <button
                  aria-label="Cancel"
                  class="cancel"
                >
                  Cancel
                </button>
              </div>
            </dialog>
          </daily-reset>
        </div>
      </div>
    </div>
  </div>
</daily-header>
`;
